src/scheduler.cc

```diff
@@ -12,18 +12,28 @@
 #include <string>
 #include <curl/curl.h>
 #include <nlohmann/json.hpp>
+#include <pthread.h>
+#include <iostream>
 
 #ifdef RCCL
   #include "rccl/rccl.h"
 #else 
   #include "nccl.h"
 #endif
-#include "parser.h"
+#include "comm.h"
```

Comment:
Pass required parameters via init() instead of header

Reply:
Need keep scheduler a standalone buildable lib

Reply:
already use passing parameter instead of header. but we can not use standalone buildable lib unless do more deep copy of the code. msccl scheduler already coupled with executor in since your last checkin.